More statistics changes and spec file integration

adrianreber opened the pull request:
https://admin.stg.fedoraproject.org/mirrormanager/propgation

pypingou commented on mirrormanager2/app.py:
```diff
@@ -1373,9 +1373,9 @@ def set_session():
-    day = date.day
+    year = date.strftime('%Y')
+    month = date.strftime('%m')
+    day = date.strftime('%d')
```
This kinda surprises me, what changed?

adrianreber replied:
Oh, the files have zero padded dates like 2015-12-05 and .day returns an integer. So that breaks. The output file is generated with strptime('%Y-%m-%d') and is thus zero padded.

pypingou replied:
ok

pypingou commented on mirrormanager2/app.py:
```diff
@@ -1436,3 +1436,39 @@ def statistics(date=None, cat='countries'):
+        prefix = 'development'
+
+    prop_base = APP.config['PROPAGATION_BASE']
+    stat_file = os.path.join(prop_base, prefix + "-repomd-propagation.svg")
```
What about having this run by ``werkzeug.secure_filename()`` to make sure it's safe?

pypingou commented on utility/mirrormanager2.spec:
```diff
@@ -134,6 +134,24 @@ Client-side, run on each downstream mirror, to report back to the
 mirror.
 
+%package statistics
+Summary:        Scripts to generate MirrorManager statistics
```
Do we have to adjust the license of the main package btw?

pypingou commented on utility/mm2_propagation:
```diff
@@ -166,42 +251,59 @@ fd.close()
+    (args.prefix, labels[len(labels) - 1].split('\n')[0]),
+    bbox_inches='tight')
+
+print("%s-%s-repomd-propagation.pdf" %
```
This is meant to be run in a cron job right? If so, should we only print when debugging?

adrianreber commented:
I updated the license, removed the debug output (if not especially requested) and the filename is now run through werkzeug.secure_filename(). Thanks for the review.

pypingou commented:
:+1: for me, do you want to patch the ansible playbook as well?
